web_external/js/views/body/UserSubmissionsView.js

```diff
@@ -0,0 +1,90 @@
+/**
+* View for a user's submissions to a phase.
+*/
+covalic.views.UserSubmissionsView = covalic.View.extend({
+    initialize: function (settings) {
+        this.phase = settings.phase;
+        this.user = settings.user;
+
+        var submissionLimit = 5;
+        if (!_.isNull(settings.submissionLimit)) {
+            submissionLimit = Math.max(1, parseInt(settings.submissionLimit, 10));
+        }
+
+        new girder.views.LoadingAnimation({
+            el: this.$el,
+            parentView: this
+        }).render();
+
+        // XXX: add way to get total number of submissions without fetching all
+        this.submissions = new covalic.collections.SubmissionCollection();
+        this.submissions.sortField = 'created';
+        this.submissions.sortDir = girder.SORT_DESC;
+        this.submissions.pageLimit = submissionLimit;
+        this.submissions.on('g:changed', function () {
+            // Assume submissions that have overallScore were successful.
+            // Fetch jobs for all other submissions.
+            var unscoredSubmissions =
+                _.filter(this.submissions.models, function (submission) {
+                    return _.isUndefined(submission.get('overallScore'));
+                });
+            if (unscoredSubmissions.length > 0) {
+                var jobs = [];
+                var promises = [];
+                _.each(unscoredSubmissions, function (submission) {
+                    var deferred = $.Deferred();
+                    var job = new girder.models.JobModel({
+                        _id: submission.get('jobId')
+                    }).on('g:fetched', function () {
+                        deferred.resolve();
+                    }, this).on('g:error', function () {
+                        deferred.reject();
+                    }, this).fetch();
+                    jobs.push(job);
+                    promises.push(deferred.promise());
+                });
+                $.when.apply($, promises).done(_.bind(function () {
```

Comment:
Nevermind...

> If a single argument is passed to jQuery.when() and it is not a Deferred or a Promise, it will be treated as a resolved Deferred and any doneCallbacks attached will be executed immediately.